ec/src/models/twisted_edwards/mod.rs

```diff
@@ -78,6 +84,63 @@ pub trait TECurveConfig: super::CurveConfig {
 
         res
     }
+
+    #[inline]
+    fn serialize_with_mode<W: Write>(
+        item: &Affine<Self>,
+        mut writer: W,
+        compress: ark_serialize::Compress,
+    ) -> Result<(), SerializationError> {
+        let flags = TEFlags::from_x_coordinate(item.x);
+        match compress {
+            Compress::Yes => item.y.serialize_with_flags(writer, flags),
+            Compress::No => {
+                item.x.serialize_uncompressed(&mut writer)?;
+                item.y.serialize_uncompressed(&mut writer)
+            },
+        }
+    }
+
+    fn deserialize_with_mode<R: Read>(
+        mut reader: R,
+        compress: Compress,
+        validate: Validate,
+    ) -> Result<Affine<Self>, SerializationError> {
+        let (x, y) = match compress {
+            Compress::Yes => {
+                let (y, flags): (_, TEFlags) =
+                    CanonicalDeserializeWithFlags::deserialize_with_flags(reader)?;
+                let (x, neg_x) = Affine::<Self>::get_xs_from_y_unchecked(y)
+                    .ok_or(SerializationError::InvalidData)?;
+                if flags.is_negative() {
+                    (neg_x, y)
+                } else {
+                    (x, y)
+                }
+            },
+            Compress::No => {
+                let x: Self::BaseField =
+                    CanonicalDeserialize::deserialize_uncompressed(&mut reader)?;
+                let y: Self::BaseField =
+                    CanonicalDeserialize::deserialize_uncompressed(&mut reader)?;
+                (x, y)
+            },
+        };
+        let point = Affine::<Self>::new_unchecked(x, y);
+        if let Validate::Yes = validate {
+            point.check()?;
+        }
+        Ok(point)
+    }
+
+    #[inline]
+    fn serialized_size(compress: Compress) -> usize {
+        let zero = Self::BaseField::zero();
+        match compress {
+            Compress::Yes => zero.serialized_size_with_flags::<TEFlags>(),
+            Compress::No => zero.uncompressed_size() + zero.uncompressed_size(),
```

Comment:
I changed this from `x.uncompressed_size() + y.uncompressed_size()` to `zero.uncompressed_size() + zero.uncompressed_size()`. Is there a difference? Aren't they all the same field? With this change this function is independent of the affine element.

Reply:
The tests passed after this change, though I'm not sure how well serialize size is tested

Reply:
It's the same right now, but if it's simpler we can keep the old behaviour.

Reply:
The new behavior is simpler. In the new behavior the size doesn't depend on the curve element, in the old behavior it referenced the curve values. So should we keep the behavior in this PR right now?